File: app/src/main/java/com/dumuzeyn/mp3player/DiscoveredTrack.java
package com.dumuzeyn.mp3player;

/** One scanned track plus its stable location inside a source tree. */
final class DiscoveredTrack {
    final Track track;
    final String documentId;
    final String identityKey;

    DiscoveredTrack(Track track, LibrarySource source, String documentId) {
        this.track = track;
        this.documentId = documentId;
        this.identityKey = TrackOrigin.identity(source.sourceId, documentId);
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/DiscoveredTrack.kt
package com.dumuzeyn.mp3player

/** One scanned track plus its stable location inside a source tree. */
class DiscoveredTrack(
    @JvmField val track: Track,
    source: LibrarySource,
    @JvmField val documentId: String,
) {
    @JvmField val identityKey: String = TrackOrigin.identity(source.sourceId, documentId)
}
